packages/snaps-sdk/src/types/methods/request-permissions.ts

```diff
@@ -0,0 +1,16 @@
+import type {
+  PermissionConstraint,
+  SubjectPermissions,
+} from '@metamask/permission-controller';
+
+/**
+ * The request parameters for the `snap_requestPermissions` method.
+ *
+ */
+export type RequestPermissionParams = Record<string, Record<string, any>>;
```

Comment:
```suggestion
export type RequestPermissionParams = RequestedPermissions;
```

Reply:
Wait, we might not want to use PermissionController as a dependency here 🤔

Reply:
Since we are trying to reduce the dependencies for snaps-sdk. WDYT @Mrtenz

Reply:
I initially wanted to use `SnapPermissions` type, but then it requires `utils` dependency, so I ended up with this temporary type. Also noticing now that `@metamask/permission-controller` might not be good as well. Not sure what would be the best to do here.

Reply:
Can we do something like this?
https://github.com/MetaMask/snaps/blob/50965668f286a0cff101f0904e2639f96331c8bf/packages/snaps-sdk/src/types/permissions.ts#L25-L62

Reply:
Using `InitialPermissions` now for the params, but the problem remains with the Result which is `SubjectPermissions<PermissionConstraint>` from Permission Controller package.

Reply:
Resolved by adding missing types from PermissionController to Snaps SDK.